core/src/main/scala/com/pingcap/tispark/v2/TiDBTableScan.scala

```diff
@@ -0,0 +1,26 @@
+package com.pingcap.tispark.v2
+
+import com.pingcap.tispark.TiTableReference
+import org.apache.spark.sql.connector.read.{Scan, Statistics, SupportsReportStatistics}
+import org.apache.spark.sql.types.StructType
+
+import java.util.OptionalLong
+
+case class TiDBTableScan(
+     tableRef: TiTableReference,
+     schema: StructType)
+  extends Scan
+    with SupportsReportStatistics {
+
+  override def readSchema(): StructType = schema
+
+  override def estimateStatistics(): Statistics = {
+    new Statistics {
+      override def sizeInBytes(): OptionalLong = {
+        val size = tableRef.sizeInBytes
+        OptionalLong.of(size)
+      }
+      override def numRows(): OptionalLong = OptionalLong.empty()
```

Comment:
Can you explain why numRows can be set as empty? Will it influence the execution plan?

if we need numRows, we can  add `numRows` in tableRef, and set the value of `numRows` in `loadStatistics` (where the sizeInBytes will be set)

Reply:
1.Can you explain why numRows can be set as empty?
Like the FileScan.class in the Spark source code, `numRows` has not been calculated. It's not necessary.

2.Will it influence the execution plan?
`sizeInBytes` can meet the most requirements in generating physical plans, but `numRows` will makes the size estimation result more accurate.

The process of size estimation can be found in SizeInBytesOnlyStatsPlanVisitor.class in the Spark source code